libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -309,7 +309,10 @@ void AP_ESC_Telem::send_esc_telemetry_mavlink(uint8_t mav_chan)
 
     // loop through groups of 4 ESCs
     const uint8_t esc_offset = constrain_int16(mavlink_offset, 0, ESC_TELEM_MAX_ESCS-1);
-    const uint8_t num_idx = ESC_TELEM_MAX_ESCS/4;
+    
+    // ensure we send out partially-full groups:
+    const uint8_t num_idx = (ESC_TELEM_MAX_ESCS % 4 == 0) ? ESC_TELEM_MAX_ESCS / 4 : (ESC_TELEM_MAX_ESCS / 4) + 1; 
```

Comment:
```suggestion
    const uint8_t num_idx = (ESC_TELEM_MAX_ESCS + 3) / 4;
```

Reply:
yes, Peters suggestion is correct

Reply:
Thanks for the review,Sure! I will rectify and update it.